src/os.rs

```diff
@@ -24,45 +28,62 @@ use Rng;
 /// - Windows: calls `RtlGenRandom`, exported from `advapi32.dll` as
 ///   `SystemFunction036`.
 /// - iOS: calls SecRandomCopyBytes as /dev/(u)random is sandboxed.
-/// - PNaCl: calls into the `nacl-irt-random-0.1` IRT interface.
 ///
 /// This usually does not block. On some systems (e.g. FreeBSD, OpenBSD,
 /// Max OS X, and modern Linux) this may block very early in the init
 /// process, if the CSPRNG has not been seeded yet.[1]
 ///
 /// [1] See <https://www.python.org/dev/peps/pep-0524/> for a more
 ///     in-depth discussion.
+#[allow(unused)]    // not used by all targets
 pub struct OsRng(imp::OsRng);
 
+impl fmt::Debug for OsRng {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
 impl OsRng {
     /// Create a new `OsRng`.
-    pub fn new() -> io::Result<OsRng> {
+    pub fn new() -> Result<OsRng, Error> {
         imp::OsRng::new().map(OsRng)
     }
 }
 
 impl Rng for OsRng {
-    fn next_u32(&mut self) -> u32 { self.0.next_u32() }
-    fn next_u64(&mut self) -> u64 { self.0.next_u64() }
-    fn fill_bytes(&mut self, v: &mut [u8]) { self.0.fill_bytes(v) }
-}
+    fn next_u32(&mut self) -> u32 {
+        // note: `next_u32_via_fill` does a byte-swap on big-endian
+        // architectures, which is not really needed here
```

Comment:
I think it was me who wrote this comment? I don't really think it matters, the system call and OS RNG should dwarf the time the byte swap takes.

Reply:
I thought it was me. But yes, I don't think it's important, so the comment could be removed.

Reply:
On that topic, `impl_uint_from_fill` does not call `.to_le()`. It should, surely?

Reply:
He, your right! While the documentations says it uses little-endian order. Really need some tests there :smile: